File: ios/runanywhere-sdks-latest/sdk/runanywhere-kotlin/src/commonMain/kotlin/com/runanywhere/sdk/core/types/AudioTypes.kt
package com.runanywhere.sdk.core

import kotlinx.serialization.Serializable

/**
 * Audio format enum matching iOS AudioFormat pattern exactly
 * This is the single source of truth for audio formats across STT, TTS, and VAD
 *
 * iOS reference: Core/Types/AudioTypes.swift
 */
@Serializable
enum class AudioFormat(
    val rawValue: String,
) {
    PCM("pcm"),
    WAV("wav"),
    MP3("mp3"),
    OPUS("opus"),
    AAC("aac"),
    FLAC("flac"),
    OGG("ogg"),
    PCM_16BIT("pcm_16bit"), // Android-specific raw PCM format
    ;

    /**
     * File extension for this format (matches iOS fileExtension)
     */
    val fileExtension: String
        get() = rawValue

    companion object {
        /**
         * Get AudioFormat from raw value string
         */
        fun fromRawValue(value: String): AudioFormat? = entries.find { it.rawValue == value.lowercase() }
    }
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-swift/Sources/RunAnywhere/Core/Types/AudioTypes.swift
//
//  AudioTypes.swift
//  RunAnywhere SDK
//
//  Audio-related type definitions used across audio components (STT, TTS, VAD, etc.)
//
//  🟢 BRIDGE: Maps to C++ rac_audio_format_enum_t
//  C++ Source: include/rac/features/stt/rac_stt_types.h
//

import CRACommons
import Foundation

// MARK: - Audio Format

/// Audio format options for audio processing
public enum AudioFormat: String, Sendable, CaseIterable {
    case pcm
    case wav
    case mp3
    case opus
    case aac
    case flac

    /// File extension for this format
    public var fileExtension: String {
        rawValue
    }

    /// MIME type for this format
    public var mimeType: String {
        switch self {
        case .pcm: return "audio/pcm"
        case .wav: return "audio/wav"
        case .mp3: return "audio/mpeg"
        case .opus: return "audio/opus"
        case .aac: return "audio/aac"
        case .flac: return "audio/flac"
        }
    }

    // MARK: - C++ Bridge (rac_audio_format_enum_t)

    /// Convert Swift AudioFormat to C++ rac_audio_format_enum_t
    public func toCFormat() -> rac_audio_format_enum_t {
        switch self {
        case .pcm: return RAC_AUDIO_FORMAT_PCM
        case .wav: return RAC_AUDIO_FORMAT_WAV
        case .mp3: return RAC_AUDIO_FORMAT_MP3
        case .opus: return RAC_AUDIO_FORMAT_OPUS
        case .aac: return RAC_AUDIO_FORMAT_AAC
        case .flac: return RAC_AUDIO_FORMAT_FLAC
        }
    }

    /// Initialize from C++ rac_audio_format_enum_t
    public init(from cFormat: rac_audio_format_enum_t) {
        switch cFormat {
        case RAC_AUDIO_FORMAT_PCM: self = .pcm
        case RAC_AUDIO_FORMAT_WAV: self = .wav
        case RAC_AUDIO_FORMAT_MP3: self = .mp3
        case RAC_AUDIO_FORMAT_OPUS: self = .opus
        case RAC_AUDIO_FORMAT_AAC: self = .aac
        case RAC_AUDIO_FORMAT_FLAC: self = .flac
        default: self = .pcm
        }
    }
}
